File: python/src/training/easy/power-of-thor-episode-1.py
import sys


def main():
    # light_x: the X position of the light of power
    # light_y: the Y position of the light of power
    # thor_x: Thor's starting X position
    # thor_y: Thor's starting Y position
    light_x, light_y, thor_x, thor_y = [int(i) for i in input().split()]

    while True:
        remaining_turns = int(input())  # The remaining amount of turns Thor can move. Do not remove this line.

        direction: str = ''
        if light_y > thor_y:
            direction = 'S'
            thor_y += 1
        else:
            if light_y < thor_y:
                direction = 'N'
                thor_y -= 1

        if light_x > thor_x:
            direction += 'E'
            thor_x += 1
        else:
            if light_x < thor_x:
                direction += 'W'
                thor_x -= 1

        # A single line providing the move to be made: N NE E SE S SW W or NW
        print(direction)


if __name__ == '__main__':
    main()

File: typescript/src/training/easy/power-of-thor-episode-1.ts
const inputs: string[] = readline().split(' ');
const lightX: number = parseInt(inputs[0]); // the X position of the light of power
const lightY: number = parseInt(inputs[1]); // the Y position of the light of power
let thorX: number = parseInt(inputs[2]); // Thor's starting X position
let thorY: number = parseInt(inputs[3]); // Thor's starting Y position

while (true) {
    const remainingTurns: number = parseInt(readline()); // The remaining amount of turns Thor can move. Do not remove this line.

    let direction = "";
    if (lightY > thorY) {
        direction = "S";
		thorY += 1;
	} else if (lightY < thorY) {
		direction = "N";
		thorY -= 1;
	}

	if (lightX > thorX) {
		direction += "E";
		thorX += 1;
	} else if (lightX < thorX) {
		direction += "W";
		thorX -= 1;
	}

    // A single line providing the move to be made: N NE E SE S SW W or NW
    console.log(direction);
}
